Nim/Nim/Nim.cpp

```diff
@@ -0,0 +1,133 @@
+using namespace std;
+
+#include <iostream>
+#include <string>
+#include <cmath>
+
+int MatchSelect(int Num, int Total, string PlayerName) // I always get the total matches left for other player from here.
+{
+    if (Num > 0 && Num < 4)
+    {
+        Total -= Num;
+    }
+    else
+    {
+        cout << "Invalid Number, you missed your chance to decide how many matches to remove from the table" << endl;
+        Num = rand() % 3 + 1;
+        Total -= Num;
+        cout << "Computer automaticly removed " << Num << " matches from the table for your turn." << endl;
+    }
+    if (Total == 0)
+    {
+        cout << "You win " << PlayerName << "' you are the winner!" << endl;
+    }
+    return Total;
+}
+
+bool GameFinished(int Total)
+{
+    if (Total >= 0) return false;
+    else return true;
+}
+
+void GameLogicVsAi()
+{
+    int Total = 24;
+    int Num;
+    string Player1;
+    string Player2 = "AI";
+
+    cout << "There are 24 matches on the table, you may draw 1, 2 or 3 matches (not more, not less) the player.\n";
+    cout << "You are playing against AI.\n";
+    cout << "Who has right to take last match from the table is going to be winner.\n";
+    cout << "Please enter your name as player 1.\n";
+    cin >> Player1;
+    while (!GameFinished(Total))
+    {
+        cout << "There are " << Total << " matches left." << endl;
+        cout << Player1 << ", please enter the amount of matches that you want to remove from the table: " << endl;
+        cin >> Num;
+
+        Total = MatchSelect(Num, Total, Player1);
+
+        if (Total == 0) break;
+        
+        if (Total <=3) //with this if clause I wanted to make sure that computer decides to finish the game while it can.
+        {
+            Num = Total;
+        }
+        else Num = rand() % 3 + 1;
+
+        cout << Player2 << ", removed "<< Num<< " matches from the table : " << endl;
+  
+
+        Total = MatchSelect(Num, Total, Player2);
+    }
+}
+
+void GameLogicVsFriend() 
```

Comment:
This method has a very high amount of duplicate code, I'm sure that you can do better than that.